k9mail-library/src/main/java/com/fsck/k9/mail/ssl/DefaultTrustedSocketFactory.java

```diff
@@ -203,6 +223,11 @@ public Socket createSocket(Socket socket, String host, int port, String clientCe
 
         setSniHost(socketFactory, sslSocket, host);
 
+        if (!proxySettings.enabled && socket == null) {
```

Comment:
I don't understand this change, and how the `socket.connect` methods below are all removed. Does the method return a connected socket now, or what?